recipes/bertrand/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+import os
+
+
+class SiConan(ConanFile):
+    name = "bertrand"
+    license = "LGPLv3"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bernedom/bertrand"
+    description = "A C++ header only library providing a trivial implementation for design by contract."
+    topics = ("design by contract", "dbc",
+              "cplusplus-library", "cplusplus-17")
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _supports_cpp17(self):
+        supported_compilers = [
+            ("gcc", "7"), ("clang", "5"), ("apple-clang", "10"), ("Visual Studio", "15.7")]
+        compiler = self.settings.compiler
+        version = Version(compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BERTRAND_BUILD_TESTING"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        elif not self._supports_cpp17():
+            raise ConanInvalidConfiguration("bertrand requires C++17 support")
```

Comment:
Please don't use this pattern. Allow unknow compilers.

Reply:
So what is the correct way to inform the user that the library is C++17 only? 
As `cppstd` is still experimental relying on it being set correctly seems a bit risky to me.

I'm checking for `cppstd` above but on a soft way
```
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, "17")
````

Reply:
Check known compilers, allow unknown compilers and display a warnging. Here is an example:
https://github.com/conan-io/conan-center-index/blob/563bf4416f2d76392ab483e5300486759567b160/recipes/spy/all/conanfile.py#L28